### PR TITLE
Jwoo/dotnet table

### PR DESCRIPTION
**Pull request by jwoo-msft**
# Related Issue
https://microsoft.visualstudio.com/OS/_workitems/edit/41839382

# Description
* Implemented Support Table
* Updated utility functions
* Updated payload tests to use set instead of list 
* Added unit tests

# How Verified

1. Unit tests were added.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/8056)

### CODE REVIEW

File: source/dotnet/Library/AdaptiveCards/AdaptiveTable.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        public AdaptiveContainerStyle GridStyleXml { get { return (GridStyle.HasValue) ? GridStyle.Value : AdaptiveContainerStyle.Default; } set { GridStyle = value; } }
+
+        /// <summary>
+        /// Determines whether to serialize the style for XML.
+        /// </summary>
+        public bool ShouldSerializeStyleXml() => this.GridStyle.HasValue;
+#endif
+
+        /// <summary>
+        /// Specifies whether the first row of the table should be treated as a header row, and be announced as such by accessibility software.
```
Comment (anna-dingler) [inline source/dotnet/Library/AdaptiveCards/AdaptiveTable.cs:92]: Is this summary supposed to be about `ShowGirdLines`?

File: source/dotnet/Library/AdaptiveCards/AdaptiveTableColumnDefinition.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+using System.Xml.Serialization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System.ComponentModel;
+
+namespace AdaptiveCards
+{
+    /// <summary>
+    /// Represents the backgroundImage property
```
Comment (anna-dingler) [inline source/dotnet/Library/AdaptiveCards/AdaptiveTableColumnDefinition.cs:12]: Comment needs updating?

File: source/dotnet/Library/AdaptiveCards/AdaptiveTableColumnDefinition.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    public class TableColumnWidth
+    {
+        public TableColumnWidth()
+        {
+        }
+
+        public TableColumnWidth(double pixelWidth)
+        {
+            PixelWidth = pixelWidth;
+        }
```
Comment (anna-dingler) [inline source/dotnet/Library/AdaptiveCards/AdaptiveTableColumnDefinition.cs:75]: nit: spacing is inconsistent here

**Review by anna-dingler — APPROVED** [review]
I left a couple of comments, but otherwise, looks good!

File: source/dotnet/Library/AdaptiveCards/AdaptiveTable.cs (hunk 1 above)
Comment (jwoo-msft) [reply source/dotnet/Library/AdaptiveCards/AdaptiveTable.cs:92]: good catch

File: source/dotnet/Library/AdaptiveCards/AdaptiveTableColumnDefinition.cs (hunk 3 above)
Comment (jwoo-msft) [reply source/dotnet/Library/AdaptiveCards/AdaptiveTableColumnDefinition.cs:75]: you mean LN62-63?

**Pull request merged**